dev/generate_helm_chart.sh

```diff
@@ -231,6 +232,7 @@ function update_deployment_resource() {
   yq e -i '.spec.template.spec.containers[1].env += ["{{ with .Values.metrics.env }}{{ toYaml . | nindent 12 }}{{ end }}"]' "${file}"
 
   perl -pi -e "s/'{{ if .Values.podAnnotations }}{{ toYaml .Values.podAnnotations \| nindent 8 }}{{ end }}': null/{{ if .Values.podAnnotations }}{{ toYaml .Values.podAnnotations \| nindent 8 }}{{ end }}/g" "${file}"
+  perl -pi -e "s/'{{- if \.Values\.operator\.priorityClassName }}priorityClassName: {{ \.Values\.operator\.priorityClassName \| quote }}{{- end }}': null/{{- if .Values.operator.priorityClassName }}priorityClassName: {{ .Values.operator.priorityClassName | quote }}{{- end }}/g" "${file}"
```

Comment:
are you sure `\.Values`  should be here according to previous strings?
how did you test it?

Reply:
why we need to use `{{\- if` here? and don't use it in `s/../..` replace part?